apps/sms/services/test/unit/messaging/messaging_service_test.js

```diff
@@ -0,0 +1,628 @@
+            success: false,
+            result: undefined,
+            code: errorResult,
+            recipient: smsOpts.recipients[idx]
+          });
+        });
+      }).then(done, done);
+    });
+
+    test('send with a serviceId in dual SIM setup', function(done) {
+      this.sinon.stub(Settings, 'hasSeveralSim').returns(true);
+      smsOpts = {
+        recipients: '123',
+        content: 'hola',
+        serviceId: 0 // we use 0 because it's falsy, to test it still works
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('send').returns(Promise.resolve(successResult));
+      MessagingService.sendSMS(smsOpts).then((results) => {
+        sinon.assert.calledWithExactly(
+          MessagingService.mozMobileMessageClient.method,
+          'send', smsOpts.recipients, smsOpts.content, { serviceId: 0 }
+        );
+      }).then(done, done);
+
+    });
+
+    test('send with a serviceId in non-dual SIM setup', function(done) {
+      this.sinon.stub(Settings, 'hasSeveralSim').returns(false);
+      smsOpts = {
+        recipients: '123',
+        content: 'hola',
+        serviceId: 1
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('send').returns(Promise.resolve(successResult));
+      MessagingService.sendSMS(smsOpts).then((results) => {
+        sinon.assert.calledWithExactly(
+          MessagingService.mozMobileMessageClient.method,
+          'send', smsOpts.recipients, smsOpts.content, undefined
+        );
+      }).then(done, done);
+
+    });
+
+    test('serviceId is a string containing a number', function(done) {
+      this.sinon.stub(Settings, 'hasSeveralSim').returns(true);
+      smsOpts = {
+        recipients: '123',
+        content: 'hola',
+        serviceId: '0'
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('send').returns(Promise.resolve(successResult));
+      MessagingService.sendSMS(smsOpts).then((results) => {
+        sinon.assert.calledWithExactly(
+          MessagingService.mozMobileMessageClient.method,
+          'send', smsOpts.recipients, smsOpts.content, { serviceId: 0 }
+        );
+      }).then(done, done);
+
+    });
+
+    test('serviceId is a bad string', function(done) {
+      this.sinon.stub(Settings, 'hasSeveralSim').returns(true);
+      smsOpts = {
+        recipients: '123',
+        content: 'hola',
+        serviceId: 'oirutoirutoitr'
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('send').returns(Promise.resolve(successResult));
+      MessagingService.sendSMS(smsOpts).then((results) => {
+        sinon.assert.calledWithExactly(
+          MessagingService.mozMobileMessageClient.method,
+          'send', smsOpts.recipients, smsOpts.content, undefined
+        );
+      }).then(done, done);
+    });
+  });
+
+  suite('sendMMS', function() {
+
+    setup(function() {
+      this.sinon.spy(SMIL, 'generate');
```

Comment:
nit: wondering if we should just stub it with predefined return result and avoid ```SMIL.generate.firstCall.returnValue```?